conan/tools/ros/rosenv.py

```diff
@@ -0,0 +1,38 @@
+import os
+from conan.api.output import Color
+from conan.tools.files import save
+
+
+class ROSEnv(object):
+    """
+    Generator to serve as integration for Robot Operating System 2 development workspaces.
+    It generates a conanrosenv.bash file that when sources sets variables so the Conan
+    dependencies are found by CMake and the run environment is also set.
+
+    IMPORTANT: This generator should be used together with CMakeDeps and CMakeToolchain generators.
+    """
+
+    def __init__(self, conanfile):
+        self._conanfile = conanfile
+        self.filename = "conanrosenv.bash"
+        self.variables = {}
+
+    def generate(self):
+        output_folder = self._conanfile.generators_folder
+        self.variables["CMAKE_TOOLCHAIN_FILE"] = os.path.join(output_folder, "conan_toolchain.cmake")
+        build_type = self._conanfile.settings.get_safe("build_type")
+        if build_type:
+            self.variables["CMAKE_BUILD_TYPE"] = build_type
+
+        content = []
+        for key, value in self.variables.items():
+            line = f"export {key}=\"{value}\""
+            content.append(line)
+        conanrun_path = os.path.join(output_folder, "conanrun.sh")
```

Comment:
Shouldn't it be the ``conanbuild.sh``?

Reply:
not sure if conanbuild.sh should be added as well.

However, remember that his conanrosenv.bash file is also setting the run environment for built package so the users only have to source one file once -> `source conanrosenv.bash`

This will set

```
export LD_LIBRARY_PATH
export DYLD_LIBRARY_PATH
export PATH
```
using `conanrun.sh` script, as this is what we need so ros generated executables can find conan installed libraries

Reply:
Oh, I see. I was thinking of cases for ``tool_requires`` that requires ``conanbuild.sh``, which is aligned to setting the "build" environment with ``CMAKE_TOOLCHAIN_FILE`` env-var, which is build-environment, not run environment.

I thought the "run" part would be injected in the ``install/setup.sh`` script, that there was a way to do that, probably I got it wrong, but I am confused.

Reply:
The mechanism to "inject" to `install/setup.sh` is weird and requires calling an ament function in the CMakeLists. of the consumer, so that's why we decided to go for the environment script approach.

Regarding tool requires, that's a valid point although I have not tested them in my examples. Maybe it should be included in the future